src/openrct2/scripting/ScEntity.hpp

```diff
@@ -77,7 +77,9 @@ namespace OpenRCT2::Scripting
             auto entity = GetEntity();
             if (entity != nullptr)
             {
+                invalidate_sprite_2(entity);
```

Comment:
Why does it need to invalidate before? (Where most only invalidate after?)

Sounds like a good case for a small RAII class

Reply:
Because the invalidate function has no way to know where it was before, making it unable to invalidate that portion of the screen.

Reply:
Hmm, I see. I wish we had used RAII here, it would be neat

Reply:
I don't think that RAII fits this use case that well. It is a very particular edge case with any kind of movement of stuff in the game, e.g. windows or entities. Ideally it should be in the sprite move function itself or rather a parameter or another method called moveandinvalidate.